File: run_sql.js
const { Client } = require('pg');
const fs = require('fs');

async function main() {
  const dbUri = "postgresql://postgres.xernedkpgdrvjokokdoa:akshitsha84@aws-1-ap-northeast-2.pooler.supabase.com:5432/postgres";
  console.log("Connecting to Supabase PostgreSQL over Node.js...");
  const client = new Client({
    connectionString: dbUri,
  });

  try {
    await client.connect();
    console.log("Connected successfully!");

    console.log("Reading full_schema_setup.sql...");
    const sqlContent = fs.readFileSync('full_schema_setup.sql', 'utf8');

    console.log("Executing SQL migrations (this will take 5-10 seconds)...");
    await client.query(sqlContent);
    console.log("🎉 SQL executed successfully and database fully seeded!");
  } catch (err) {
    console.error("Error executing SQL:", err);
    process.exit(1);
  } finally {
    await client.end();
  }
}

main();

File: run_sql.py
import psycopg2
import sys

def main():
    # Connecting using Supabase Seoul connection pooler
    db_uri = "postgresql://postgres.xernedkpgdrvjokokdoa:akshitsha84@aws-0-ap-northeast-2.pooler.supabase.com:6543/postgres"
    print("Connecting to Supabase Seoul connection pooler...")
    try:
        conn = psycopg2.connect(db_uri)
        conn.autocommit = True
        cursor = conn.cursor()
        print("Connected successfully!")
        
        print("Reading full_schema_setup.sql...")
        with open("full_schema_setup.sql", "r", encoding="utf-8") as f:
            sql_content = f.read()
            
        print("Executing schema and seeding query (this will take 5-10 seconds)...")
        cursor.execute(sql_content)
        print("SQL executed successfully and database fully seeded!")
        
        cursor.close()
        conn.close()
    except Exception as e:
        print(f"Error executing SQL: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()
